### PR TITLE
Interpolation -> KA kernels + Variable polynomial orders.

### PR DESCRIPTION
**Pull request by skandalaCLIMA**
1). Replaced separate CUDA and CPU kernels with KernelAbstractions kernels.

2). Enabled support for different polynomial orders in different
directions.

### Description

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by thomasgibson** [comment]
Thanks @skandalaCLIMA! If this is pretty self-contained, then it's probably possible to go ahead and clean/merge this once you think it's ready. This doesn't seem to rely on anything in #1702 --- so they can probably be merged independently

File: src/Numerics/Mesh/Interpolation.jl [hunk 1]
```diff
@@ -739,8 +703,8 @@ struct InterpolationCubedSphere{
         poly_order = polynomialorders(grid)
         # Currently only support single polynomial order
         @assert all(poly_order[1] .== poly_order)
-        poly_order = poly_order[1]
-        qm1 = poly_order + 1
+#        poly_order = poly_order[1]
```
Comment (kpamnany) [inline src/Numerics/Mesh/Interpolation.jl:706]:
```suggestion
```

**Comment by skandalaCLIMA** [comment]
bors try

**Comment by skandalaCLIMA** [comment]
bors try

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**